source/openpulse/openpulse/printer.py

```diff
@@ -0,0 +1,76 @@
+import io
+
+from openqasm3.printer import Printer as QASMPrinter
+from openqasm3.printer import PrinterState
+
+import openpulse.ast as ast
+
+
+def dump(node: ast.QASMNode, file: io.TextIOBase, **kwargs) -> None:
+    """Write textual OpenQASM 3 code representing ``node`` to the open stream ``file``.
+
+    It is generally expected that ``node`` will be an instance of :class:`.ast.Program`, but this
+    does not need to be the case.
+
+    For more details on the available keyword arguments, see :ref:`printer-kwargs`.
+    """
+    Printer(file, **kwargs).visit(node)
+
+
+def dumps(node: ast.QASMNode, **kwargs) -> str:
+    """Get a string representation of the OpenQASM 3 code representing ``node``.
+
+    It is generally expected that ``node`` will be an instance of :class:`.ast.Program`, but this
+    does not need to be the case.
+
+    For more details on the available keyword arguments, see :ref:`printer-kwargs`.
+    """
+    out = io.StringIO()
+    dump(node, out, **kwargs)
+    return out.getvalue()
+
+
+class Printer(QASMPrinter):
```

Comment:
OpenPulse `Printer` is a straightforward subclass of `openqasm3.printer.Printer`.